odw_common/util/synapse_workspace_manager.py

```diff
@@ -0,0 +1,169 @@
+                    "upload",
+                    "--workspace-name",
+                    self.workspace_name,
+                    "--package",
+                    package_path,
+                    "--no-progress"
+                ]
+            )
+        )
+        package_name = package_path.replace("dist/", "")
+        max_wait_time = 5 * 60 # Wait 5 minutes
+        current_wait_time = 0
+        retry_delay_seconds = 20
+        while current_wait_time < max_wait_time:
+            workspace_package_names = {package["name"]: package for package in self.get_workspace_packages()}
+            if package_name in workspace_package_names:
+                if workspace_package_names[package_name]["properties"]["provisioningStatus"] == "Succeeded":
+                    return resp
+            current_wait_time += retry_delay_seconds
+            time.sleep(retry_delay_seconds)
+        raise MaxWaitTimeNeededException(f"Exceeded max wait time for creation of workspace package '{package_name}'")
+
+    def remove_workspace_package(self, package_name: str):
+        """
+            Remove the given package from the Synapse workspace
+
+            :param package_name: The name of the package to remove. e.g: `some_python_wheel.whl`
+        """
+        # This command returns nothing
+        Util.run_az_cli_command(
+            [
+                "az",
+                "synapse",
+                "workspace-package",
+                "delete",
+                "--workspace-name",
+                self.workspace_name,
+                "--package",
+                package_name,
+                "--no-wait",
+                "-y"
+            ]
+        )
+        max_wait_time = 10 * 60 # Wait 10 minutes
+        current_wait_time = 0
+        retry_delay_seconds = 20
+        while current_wait_time < max_wait_time:
+            workspace_package_names = [package["name"] for package in self.get_workspace_packages()]
+            if package_name not in workspace_package_names:
+                return
+            current_wait_time += retry_delay_seconds
+            time.sleep(retry_delay_seconds)
+        raise MaxWaitTimeNeededException(f"Exceeded max wait time for deletion of workspace package '{package_name}'")
+
+    def get_spark_pool(self, spark_pool_name: str) -> Dict[str, Any]:
+        """
+            Return the json for the given spark pool. An exception is raised if there is an error with the request
+
+            :param spark_pool_name: The name of the spark pool to fetch.
+        """
+        resp = requests.Response = requests.get(
+            f"{self.ENDPOINT}/bigDataPools/{spark_pool_name}?api-version=2021-06-01",
+            headers={"Authorization": f"Bearer {self._get_token()}"}
+        )
+        try:
+            resp = resp.json()
+            if "error" in resp:
+                raise ValueError(f"A http exception was raised when calling get_spark_pool(): {json.dumps(resp, indent=4)}")
```

Comment:
Isn't this a bug ? You assign to the requests.Response class resp = requests.Response = requests.get(....) Shouldn't it be :
response = requests.get(....); data = response.json() and then check response.status_code